### ISSUE TITLE
Implementing CBOW (Bag of words) in tflearn+google cloud ML engine

### ISSUE THREAD

**Issue by shauravg** [tflearn/tflearn#851]
I need to create a model which accepts a user utterance (string), stems each word and then compares it against the array of known words to convert into a bag of words array and then pass it through a DNN model. Example:

1. user says "Hello, how are you?"
2. Create a stemmed word array. Let's say: [hello, how, are, you]
3. known words array is [hello]
4. bag of words array is [1 0 0 0]
5. from here, I know I can do something like this:

net = tflearn.input_data(shape=[None, len(train_x[0])]) 
net = tflearn.fully_connected(net, 4)
net = tflearn.fully_connected(net, 4)
net = tflearn.fully_connected(net, len(train_y[0]), activation='softmax')
net = tflearn.regression(net)
model = tflearn.DNN(net, tensorboard_dir='somedir')
model.fit(train_x, train_y, n_epoch=500, batch_size=4)

But I need the input_data to accept dtype=tf.string and then use another layer/op to do the above operations so that I can deploy to Google cloud ML engine. It is working just fine if I do the above operations in separate python functions but I need to deploy to google cloud ml engine and then be able to invoke the prediction interface from a node.js server.
